book/src/multithreading.md

```diff
@@ -0,0 +1,45 @@
+# Multithreading
+
+RIOT-rs supports multithreading on the Cortex-M, RISC-V, and Xtensa architectures.
+
+## Scheduling
+
+- **Fixed-priority, preemptive scheduling** policy with up to 32 supported priority levels.
+- **Thread priorities are dynamic** and can be changed at runtime.
+- The Runqueue is implemented as a circular linked list for each priority, requiring n × m × 8 bits for _n_ maximum threads and _m_ supported priorities. All operations on the runqueue are performed in constant time, except for the deletion of a thread that is not the head of the runqueue.
```

Comment:
This is *very* detailed, too detailed for this overview IMO. Mentioning the `O(1)` operations make sense though. (when is  a thread that is not head deleted again?)

Reply:
> when is  a thread that is not head deleted again?

When the priority of the thread is changed.